src/sparse_merkle_tree/smt.rs

```diff
@@ -842,38 +842,78 @@ impl SparseMerkleTree {
         Ok((query_with_proofs, ancestor_hashes))
     }
 
-    fn verify_query_keys(proof: &Proof, query_keys: &[Vec<u8>], key_length: KeyLength) -> bool {
-        let mut queries: HashMap<&[u8], QueryProof> = HashMap::new();
+    /// verify_and_prepare_proof_map checks all verifications of query_keys based on the verify function in the [LIP-0039](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0039.md#proof-construction).
+    /// Also it returns a tuple, the first parameter is the result of the query_keys verification and the second one is a map
+    fn verify_and_prepare_proof_map(
+        proof: &Proof,
+        query_keys: &[Vec<u8>],
+        key_length: KeyLength,
+    ) -> Result<HashMap<Vec<bool>, QueryProofWithProof>, SMTError> {
+        let mut queries_with_proof: HashMap<Vec<bool>, QueryProofWithProof> = HashMap::new();
```

Comment:
it will initialize by `HashMap::new()`, you mean, I can initialize it by specific data?

Reply:
@ricott1 probably meant that line 852 could be moved down to line 859. Speed optimization, unnecessary initialization of hash map in case that if statement returns from function.

Reply:
Done, sorry for my misunderstanding.